src/entities/field.py

```diff
@@ -0,0 +1,155 @@
+        goals = []
+        for goal in goal_dict:
+            color = goal["color"]         
+
+            pose = self.__parse_position(goal["position"])
+
+            ring_containers = self.__parse_ring_containers(goal["ring_containers"])
+
+            tipped = goal["tipped"]
+
+            if color == Color.RED:
+                goals.append(RedGoal(pose, ring_containers=ring_containers, tipped=tipped))
+            elif color == Color.BLUE:
+                goals.append(BlueGoal(pose, ring_containers=ring_containers, tipped=tipped))
+            else:
+                goals.append(NeutralGoal(pose, ring_containers=ring_containers, tipped=tipped))
+
+        return goals
+
+    def __parse_robots(self, robots_dict: dict) -> Robot:
+        robots = []
+        for robot in robots_dict:
+            color = robot["color"]
+
+            id = robot["id"]
+
+            pose = self.__parse_position(robot["position"])
+
+            rings = self.__parse_ring_containers(robot["rings"])
+
+            goals = self.__parse_goals(robot["goals"])
+
+            tipped = robot["tipped"]
+
+            if id == RobotID.SELF:
+                robots.append(HostRobot(color, pose, rings=rings, goals=goals, tipped=tipped))
+            elif id == RobotID.PARTNER:
+                robots.append(PartnerRobot(color, pose, rings=rings, goals=goals, tipped=tipped))
+            else:
+                robots.append(OpposingRobot(color, pose, rings=rings, goals=goals, tipped=tipped))
+
+        return robots
+
+    def __parse_platform(self, platform_dict: dict) -> Platform:
+        color = platform_dict["color"]
+
+        goals = self.__parse_goals(platform_dict["goals"])
+
+        rings = self.__parse_rings(platform_dict["rings"])
+
+        robots = self.__parse_robots(platform_dict["robots"])
+
+        state = platform_dict["state"]
+
+        if color == Color.RED:
+            return RedPlatform(state, rings=rings, goals=goals, robots=robots)
+        else:
+            return BluePlatform(state, rings=rings, goals=goals, robots=robots)
+
+    def parse_representation(self, representation: str) -> None:
+        self.rings = self.__parse_rings(representation["rings"])
+
+        self.goals = self.__parse_goals(representation["goals"])
+
+        self.red_platform = self.__parse_platform(representation["red_platform"])
+        
+        self.blue_platform = self.__parse_platform(representation["blue_platform"])
+
+        self.robots = self.__parse_robots(representation["robots"])
+
+    def as_json(self) -> str:
+        return json.dumps(self, default=lambda o: o.__dict__, sort_keys=True, indent=4)
+
+
+class FieldState(ISerializable):
+    def __init__(self, representation: Field, time: int):
+        self.potential_score = (0, 0)
+        self.current_time = time
+        self.field_representation = representation
+
+    def __calculate_potential_score(self) -> None:
+        return None  # TODO
```

Comment:
Agent rewarding is encapsulated in the gym env, but this could be an accessor for either current points or scoring elements prioritized by a heuristic like distance.

Reply:
Definitely. I left this as a TODO as we may want to return different aspects of scoring separately (ex. points in goals, points in rings, etc.) to allow us to weight them differently.